source/kraken/apply_disruption.cpp

```diff
@@ -424,21 +472,35 @@ struct add_impacts_visitor : public apply_impacts_visitor {
             } else {
                 // Affect the headsign to vj if present in gtfs-rt
                 if (!impact->headsign.empty()) {
-                    vj->headsign = impact->headsign;
                     vj->name = impact->headsign;
-                    pt_data.headsign_handler.change_name_and_register_as_headsign(*vj, impact->headsign);
+                    pt_data.headsign_handler.change_vj_headsign_and_register(*vj, impact->headsign);
                 }
 
+                // Affect the trip_short_name to vj if present in gtfs-rt
+                if (!impact->trip_short_name.empty()) {
+                    vj->name = impact->trip_short_name;
+                }
+
+                nt::Dataset* dataset = nullptr;
+                if (!impact->dataset_id.empty()) {
+                    dataset = pt_data.get_dataset(impact->dataset_id);
+                    if (dataset == nullptr) {
+                        LOG4CPLUS_INFO(log, "Problem with impact (" << impact->uri << "): dataset_id ("
+                                                                    << impact->dataset_id << ") provided is absent");
+                    }
+                }
                 // for protection, use the datasets[0]
                 // TODO : Create default data set
-                if (!pt_data.datasets.empty()) {
-                    if (pt_data.datasets[0]) {
-                        vj->dataset = pt_data.datasets[0];
-                        vj->dataset->vehiclejourney_list.insert(vj);
-                    }
+                if (dataset == nullptr && !pt_data.datasets.empty() && pt_data.datasets[0]) {
+                    dataset = pt_data.datasets[0];
+                    LOG4CPLUS_INFO(log, "impact ("
+                                            << impact->uri
+                                            << "): associate random dataset to new VJ because base VJ doesn't exist");
+                }
+                if (dataset != nullptr) {
+                    vj->dataset = dataset;
+                    vj->dataset->vehiclejourney_list.insert(vj);
```

Comment:
Can this block be moved above and be include in 
if (dataset == nullptr && !pt_data.datasets.empty() && pt_data.datasets[0]) {

Reply:
Letting it here allows checking that dataset is not `nullptr` after previous block (in case `pt_data.datasets[0]`is null).
In this case, this line would crash.
So really unlikely, but protecting a crash was the goal here.

Reply:
:+1: